_git/git_commands.py

```diff
@@ -0,0 +1,434 @@
+from aenea import Text
+
+'''
+NOTE: Not all possible commands and options are available in this grammar.
+NatLink puts a limit on how complex the grammar can get, and some commands had
+to be removed. If you feel like there is something missing here, feel free to
+add stuff back and make a pull request.
+
+NOTE 2: Lists of options were grepped from the Git help pages using a bash
+script. These have been marked with the comment 'Generated'. Some of these
+generated lists are too long, so had to be shortened (see note above). You can
+find the script here in case you find it useful
+https://gist.github.com/dylan-chong/ecf701b1a623c9f2ccb78cbb5db700c6 . Ideally,
+when adding a new command, just add the options that are commonly used.
+'''
+
+
+# Common refs for convenience. The user will still have to type out most branch
+# and remote names themselves
+_COMMON_BRANCH_NAMES = ['master', 'develop', 'HEAD']
+_COMMON_REMOTE_NAMES = ['origin', 'upstream']
+
+
+def _add_common_refs(rule_builder):
+    for remote in _COMMON_REMOTE_NAMES:
+        for branch in _COMMON_BRANCH_NAMES:
+            rule_builder.smart_options([remote + '/' + branch])
```

Comment:
Note that this is not a particularly scalable way of adding new branch names. As you add new branch all remote names one by one, there is an exponential increase in how many options there are.

But I don't think there *that* are many common branch and remote names, so I think this simple implementation is okay

Reply:
Seems likely fine by me, though I might consider adding a comment to this effect in the event that someone is trying to troubleshoot slow recognition.

Reply:
I didn't experience any slow recognition when I had too many options. I encountered too complex grammar error at one point, but when I remove just enough options to get rid of this error, recognition speed seem to be the same as usual.